src/lib/db/migrations/20230331084321_processed-blocks-table.ts

```diff
@@ -0,0 +1,20 @@
+import { Knex } from 'knex'
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable('processed_blocks', (def) => {
+    def.specificType('hash', 'CHAR(64)').notNullable()
+    def.bigInteger('height').unsigned().notNullable().unique()
+    def.specificType('parent', 'CHAR(64)').notNullable()
+    def.datetime('created_at').notNullable().defaultTo(knex.fn.now())
+
+    def.primary(['hash'])
+  })
+
+  await knex.schema.alterTable('processed_blocks', (def) => {
+    def.foreign('parent', 'fk_processed_blocks_parent_hash').references('hash').inTable('processed_blocks')
```

Comment:
how would the first insert work? Seems like a chicken/egg thing, you need an existing row with a hash to set as parent, and that row needs a row for its parent and so on...

Reply:
since the columns are `NotNullable`

Reply:
A row can reference itself